pkg/operator/apiservicecontroller/apiservice_controller.go

```diff
@@ -230,3 +248,111 @@ func (c *APIServiceController) eventHandler() cache.ResourceEventHandler {
 		DeleteFunc: func(obj interface{}) { c.queue.Add(workQueueKey) },
 	}
 }
+
+// externalOperatorPreconditionFunc represents a function that will be called by this operator
+// to determine if an external operator is ready to take over some API services
+type externalOperatorPreconditionFuncType func() error
+
+// GetAPIServicesToManageWrapper returns a function that returns the desired list of API Services that will be managed by this operator
+// note that some services might be managed by an external operators/servers
+func GetAPIServicesToManageWrapper(apiregistrationv1Client apiregistrationv1client.ApiregistrationV1Interface,
+	operatorPrecondition externalOperatorPreconditionFuncType,
+	authoritativeAPIServicesList []*apiregistrationv1.APIService,
+	eventRecorder events.Recorder,
+	apiGroupsManagedByExternalServer sets.String,
+	apiGroupsManagedByExternalServerAnnotation string) func() ([]*apiregistrationv1.APIService, error) {
+
+	observedAPIServices := authoritativeAPIServicesList
+	return func() ([]*apiregistrationv1.APIService, error) {
+		if externalOperatorPrecondition := operatorPrecondition(); externalOperatorPrecondition != nil {
+			return nil, externalOperatorPrecondition
```

Comment:
we could return `authoritativeAPIServicesList` when the operator is not available.

Reply:
> we could return `authoritativeAPIServicesList` when the operator is not available.

yeah, that allows coasting.